backend/sozisel/lib/sozisel/model/event_results.ex

```diff
@@ -47,4 +50,116 @@ defmodule Sozisel.Model.EventResults do
   def change_event_result(%EventResult{} = event_result, attrs \\ %{}) do
     EventResult.create_changeset(event_result, attrs)
   end
+
+  def get_all_event_results(%LaunchedEvent{} = launched_event) do
+    launched_event
+    |> Repo.preload(:event_results)
+    |> then(& &1.event_results)
+  end
+
+  def quiz_summary(%LaunchedEvent{} = event_launched) do
+    quiz_results = get_all_event_results(event_launched)
+
+    total_points =
+      quiz_results
+      |> Enum.flat_map(& &1.result_data.participant_answers)
+      |> Enum.map(& &1.points)
+      |> Enum.sum()
+
+    total_quiz_answer_time =
+      quiz_results |> Enum.map(& &1.result_data.quiz_answer_time) |> Enum.sum()
+
+    participant_answers =
+      quiz_results
+      |> Enum.flat_map(& &1.result_data.participant_answers)
+
+    %{
+      number_of_participants: length(quiz_results),
+      average_points: total_points / length(participant_answers),
+      average_quiz_answer_time: total_quiz_answer_time / length(quiz_results)
+    }
+  end
+
+  def quiz_participants_summary(%LaunchedEvent{} = event_launched) do
+    quiz_results = get_all_event_results(event_launched)
+
+    quiz_participant_summary =
+      quiz_results
+      |> Enum.map(fn participant_result ->
+        result_data =
+          participant_result
+          |> Utils.from_deep_struct()
+          |> Map.get(:result_data)
+
+        participant_answers =
+          result_data
+          |> Map.get(:participant_answers)
+
+        number_of_points =
+          participant_answers
+          |> Enum.map(& &1.points)
+          |> Enum.sum()
+
+        %Participant{email: email, full_name: full_name} =
+          Participants.get_participant!(Map.get(participant_result, :participant_id))
+
+        %{
+          full_name: full_name,
+          email: email,
+          number_of_points: number_of_points,
+          quiz_answer_time: result_data |> Map.get(:quiz_answer_time),
+          participant_answers: participant_answers
+        }
+      end)
+
+    quiz_participant_summary
+  end
+
+  def quiz_questions_summary(%LaunchedEvent{} = event_launched) do
+    quiz_results = get_all_event_results(event_launched)
+
+    event = Events.get_event(event_launched.event_id)
```

Comment:
Can you try something like this? `launched_event |> Ecto.assoc(:event) |> Repo.one()`? I have no idea if this will work but will look much cleaner I guess